### PR TITLE
[plugin.video.s04tv] 3.1.8

### PR DESCRIPTION
**Pull request by maloep**
### Description
- Addon to watch videos from FC Schalke 04 online video portal.

### Checklist:

- [X ] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [ X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [ X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.s04tv/default.py [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+except:
+    pass
+try:
+    mode=int(params["mode"][0])
+except:
+    pass
+
+print "Mode: "+str(mode)
+print "URL: "+str(url)
+print "Name: "+str(name)
```
Comment (Rechi) [inline plugin.video.s04tv/default.py]: use xbmc.log or remove the lines

File: plugin.video.s04tv/addon.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <language>de</language>
+        <platform>all</platform>
+        <license>GNU GENERAL PUBLIC LICENSE. Version 2, June 1991</license>
+        <forum>http://forum.kodi.tv/showthread.php?tid=109692</forum>
+        <website>https://github.com/maloep/xbmc-s04tv</website>
+        <source>https://github.com/maloep/xbmc-s04tv</source>
+        <summary lang="de">Videos von S04.tv ansehen</summary>
```
Comment (enen92) [inline plugin.video.s04tv/addon.xml]: de_DE

File: plugin.video.s04tv/addon.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        <summary lang="en">Watch videos from S04.tv</summary>
```
Comment (enen92) [inline plugin.video.s04tv/addon.xml]: en_GB

File: plugin.video.s04tv/addon.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        <description lang="de">Online Video Portal des FC Schalke 04. Alle Spiele in re-Live (15 Minuten nach Abfiff), News, Interviews, exklusive Berichte und Specials. Kostenlos für Schalke-Mitglieder, die re-Live Spiele setzen allerdings einen S04 TV Benutzer Account voraus (19,04 EUR/Jahr). Mehr Details auf https://schalke04.de/tv.</description>
+        <description lang="en">FC Schalke 04 online video portal. Watch all videos from S04tv.de including all matches in re-Live (15 minutes after the match has ended), news, interviews and specials. Free of charge for S04 members, re-live videos require an S04 TV user account (19,04 EUR/year). See https://schalke04.de/tv for details.</description>
```
Comment (enen92) [inline plugin.video.s04tv/addon.xml]: same as the 2 above

File: plugin.video.s04tv/default.py [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+        if not has_schalke_tv:
+            xbmc.log("This video requires payed Schalke TV subscription")
+            xbmcgui.Dialog().ok(PLUGINNAME, language(30104), language(30105))
+            return
+
+    
+    #HACK: Free content may be hosted on youtube
+    if(url.startswith("https://youtu.be/")):
+        videoId = url.replace("https://youtu.be/", "")
+        url='plugin://plugin.video.youtube/?action=play_video&videoid=' +videoId
```
Comment (enen92) [inline plugin.video.s04tv/default.py]: plugin://plugin.video.youtube/play/?video_id=$VIDEOID

File: plugin.video.s04tv/default.py [hunk 6]
```diff
@@ -0,0 +1,369 @@
...
+            
+    listitem = xbmcgui.ListItem(path=videourl)
+    return xbmcplugin.setResolvedUrl(thisPlugin, True, listitem)
+    
+
+def addDir(name, url, mode, iconimage):
+    parameters = {'url' : url.encode('utf-8'), 'mode' : str(mode), 'name' : name.encode('utf-8')}
+    u = sys.argv[0] +'?' +urllib.urlencode(parameters)
+    ok = True
+    listitem = xbmcgui.ListItem(name, iconImage="DefaultFolder.png", thumbnailImage=iconimage)
```
Comment (enen92) [inline plugin.video.s04tv/default.py]: iconImage and thumbnailImage are both deprecated, please use listitem.setArt{"icon":"xxx", "thumb":"..."}

File: plugin.video.s04tv/default.py [hunk 7]
```diff
@@ -0,0 +1,369 @@
...
+    listitem.setInfo(type="Video", infoLabels={"Title": name})
+    ok = xbmcplugin.addDirectoryItem(thisPlugin, u, listitem, isFolder=True)
+    return ok
+   
+
+def addLink(name, url, mode, iconimage, date, extraInfo):
+    parameters = {'url' : url.encode('utf-8'), 'mode' : str(mode), 'name' : name.encode('utf-8')}
+    u = sys.argv[0] +'?' +urllib.urlencode(parameters)
+    ok = True
+    listitem = xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
```
Comment (enen92) [inline plugin.video.s04tv/default.py]: same as above

File: plugin.video.s04tv/default.py [hunk 8]
```diff
@@ -0,0 +1,369 @@
...
+def searchMemberStatus(soup, status):
+    loginSuccessful = False
+    
+    for textelement in soup(text=status):
+        ahref = textelement.parent
+        li = ahref.parent
+        try:
+            isChecked = li['class'] == 'checked'
+            loginSuccessful = isChecked
+        except KeyError:
```
Comment (enen92) [inline plugin.video.s04tv/default.py]: same as rechis comment

File: plugin.video.s04tv/resources/language/English/strings.xml [hunk 9]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (enen92) [inline plugin.video.s04tv/resources/language/English/strings.xml]: xml format is seriously deprecated. Can you move to strings.po?
Also the folder layout should be `/resources/language/resource.language.en_gb/strings.po`

File: plugin.video.s04tv/resources/language/German/strings.xml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
Comment (enen92) [inline plugin.video.s04tv/resources/language/German/strings.xml]: same as above but with de_DE

File: plugin.video.s04tv/default.py (hunk 8 above)
Comment (Rechi) [reply plugin.video.s04tv/default.py]: this is py3 compatible

**Comment by enen92** [comment]
Thanks much

**Pull request merged**